packages/video_player/video_player/lib/video_player.dart

```diff
@@ -519,6 +532,43 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     await _videoPlayerPlatform.setVolume(_textureId, value.volume);
   }
 
+  /// Returns true if picture in picture is supported on the device.
+  Future<bool> isPictureInPictureSupported() {
+    return _videoPlayerPlatform.isPictureInPictureSupported();
+  }
+
+  /// Prepare picture in picture by passing the location of the video player view
+  Future<void> preparePictureInPicture({
+    double top = 0,
+    double left = 0,
+    double width = 0,
+    double height = 0,
+    bool enableStartPictureInPictureAutomaticallyFromInline = false,
+  }) async {
+    if (!value.isInitialized || _isDisposed) {
+      return;
+    }
+    await _videoPlayerPlatform.preparePictureInPicture(
+      textureId: _textureId,
+      enableStartPictureInPictureAutomaticallyFromInline:
+          enableStartPictureInPictureAutomaticallyFromInline,
+      top: top,
+      left: left,
+      width: width,
+      height: height,
+    );
+  }
+
+  /// Start/stop picture in picture mode
+  Future<void> setPictureInPicture(
```

Comment:
This "set" name isn't clear that this will actually start or stop the PiP.  `toggleOnOrOff`? I can't think of a great name.

Reply:
Maybe I should switch to 2 methods.
`startPictureInPicture` and `stopPictureInPicture?